build.py

```diff
@@ -0,0 +1,229 @@
+#!/usr/bin/python
+
+import optparse
+import glob
+import codecs
+import os
+import shutil
+import logging
+
+import markdown
+from Cheetah.Template import Template
+from git import Repo
+import yaml
+
+logging.basicConfig(level=logging.DEBUG)
+
+
+class DocBuilder(object):
+
+    def __init__(self, config, versions):
+        self.config = config
+        self.versions = versions
+
+    def run(self):
+        '''
+        Main entry point for building the docs.
+        '''
+        for v in self.versions:
+            logging.info('Building docs for %s', v)
+            try:
+                self.build(v, 'en')
+            except:
+                logging.exception('skipping version "%s" due to error', v)
+                continue
+
+    def build(self, version, lang, doc=None):
+        '''
+        Handle building a single version of the docs.
+        '''
+        try:
+            self.repo = Repo('.')
+        except:
+            logging.exception('unable to find git repository')
+        original_ref = self.repo.head.reference
+        try:
+            # Grab the correct version.
+            assert not self.repo.bare
+            assert not self.repo.is_dirty()
```

Comment:
I agree; there's already validation of repo status (dirty or not) in the `checkout` function, so these asserts aren't really needed here. I'll fix.